### DISCUSSION TITLE
I don't see request body/data when I am using axios GET method in browser but I see in server/node.

### DISCUSSION THREAD

**Discussion by mar1n** [mswjs/msw#1695] [General]
I know that Axios dropped sending body/data for GET method.
So when I do in handler:
`const { firstName } = await req.json();  `
This doesn't work in browser.
I have this error message:
SyntaxError: Unexpected end of JSON input
But when I am lunching test in server/node environment(JEST) everything is OK I am receiving body/data.
Does any know why?

**Comment by kettanaito** [comment]
Hi, @mar1n. Can you please provide some additional context around the reproduction of this behavior?

- What version of MSW are you using?
- What version of Axios are you using?
- Do you have a chance to set up a reproduction repository or sandbox for this?

I think the difference you experience may be related to how Axios works in the browser and Node.js. In the browser, Axios uses `XMLHttpRequest` to represent requests, while in Node.js it uses `http.ClientRequest`. Those are two completely different request implementations that don't follow the same specification. This may be the reason why the behavior differs for you.

**Reply by mar1n** [reply]
- MSW version: 0.44.2
- Axios: 0.27.2
Your answer is **enough** for me I suspected that node is using something else.
I will stop using in Axios/Get body/data and will changed to parameters.

https://stackoverflow.com/questions/978061/http-get-with-request-body
